docs/application/dotnet/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,607 @@
+
+To add and manage widget components, you must add a widget component and register the widget component object.
+
+1. Declare a widget component in the manifest file:
+    ```
+    <component-based-application appid="org.tizen.base-component" exec="@BINDIR@/base-component" nodisplay="false" multiple="false" type="dotnet">
+        <label>Base-component application</label>
+        <icon>@DESKTOP_ICON@</icon>
+        <widget-component id="base-widget" setup-appid="base-widget" max-instance=0>
+            <icon>org.tizen.sample.png</icon>
+            <label>base-widget</label>
+            <label xml:lang="en-us">base-widget</label>
+            <label xml:lang="ko-kr">base-widget[KOR]</label>
+            <support-size>2x2</support-size>
+        </widget-component>
+    </component-based-application>
+    ```
+
+2. Add a widget component to the component based application:
+    ```
+    public class WidgetComp : WidgetComponent
+    {
+        internal static readonly string LogTag = "CompBasedCsharp";
+        MyWindowProxy win_;
+
+        public override IWindowProxy CreateWindowInfo(int width, int height)
+        {
+            Window window = new Window("base-widget")
+            win_ = new MyWindowProxy(window);
+            return win_;
+        }
+        public override bool OnCreate(int width, int height)
+        {
+            Log.Debug(LogTag, "OnCreate Widget");
+            return true;
+        }
+        public override void OnStart(bool restarted)
+        {
+            Log.Debug(LogTag, "OnStart Widget");
+        }
+        public override void OnResume()
+        {
+            Log.Debug(LogTag, "OnResume Widget");
+            base.OnResume();
+        }
+        public override void OnPause()
+        {
+            Log.Debug(LogTag, "OnPause Widget");
+            base.OnPause();
+        }
+        public override void OnDestroy(bool permanent)
+        {
+            Log.Debug(LogTag, "OnDestroy Widget");
+            base.OnDestroy(permanent);
+        }
+        public override void OnStop()
+        {
+            Log.Debug(LogTag, "OnStop Widget");
+            base.OnStop();
+        }
+    }
+
+    namespace CompBasedCsharp
+    {
+        internal class App : EFLComponentBasedApplication
+        {
+            internal static readonly string LogTag = "CompBasedCsharp";
+            App(IDictionary<Type, string> typeInfo) : base(typeInfo)
+            {
+            }
+
+            static void Main(string[] args)
+            {
+                Elementary.Initialize();
+                Elementary.ThemeOverlay();
+                Log.Error(LogTag, "start");
+                IDictionary<Type, string> typeInfo = new Dictionary<Type, string>();
+                typeInfo.Add(typeof(WidgetComp), "base-widget");
+                App app = new App(typeInfo);
+                app.Run(args);
+            }
+        }
+    }
+    ```
+
+<a name="launch_application"></a>
+## Launch Component Based Application
```

Comment:
All the headings, H1, H2, H3 need to be in sentence case

Reply:
I didn't understand this comment exactly,
but I added two openings at all headings.